### PR TITLE
Complaint if Joint doesn't exists before adding joint controller

### PR DESCRIPTION
**Pull request by cjds**
# 🎉 New feature

Closes :  #<NUMBER> Sorry haven't written an issue yet

## Summary

- Changed the JointController class to inform user when the joint trying to be controlled doesn't exist on the model
- Changed it similarly to JointStatePublisher to show similar error message1

## Test it
Tested this locally by trying to control joints that didn't exist in a model

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/systems/joint_controller/JointController.cc [hunk 1]
```diff
@@ -98,6 +98,13 @@ void JointController::Configure(const Entity &_entity,
     return;
   }
 
+  gazebo::Entity jointEntity = this->dataPtr->model.JointByName(_ecm, this->dataPtr->jointName);
+  if (jointEntity == kNullEntity)
+  {
+    ignerr << "Joint with name[" << this->dataPtr->jointName << "] not found. "
+    << "The JointController will not control this joint.\n";
+  }
```
Comment (ahcorde) [inline src/systems/joint_controller/JointController.cc:106]: if this an error we should probably return here

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/systems/joint_controller/JointController.cc (hunk 1 above)
Comment (cjds) [reply src/systems/joint_controller/JointController.cc:106]: @ahcorde I'm open to changing it to an error. Could you explain how to do thaat?

I based this on https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo5/src/systems/joint_state_publisher/JointStatePublisher.cc#L75 which doesn't display an error but just prints to cerr as well

**Comment by chapulina** [comment]
There's some logic on `PreUpdate` to continuously look for the joint if it doesn't exist yet. This suggests to me that in some use cases, the joint may not be present on the model yet when `Cofigure` is called, but it may show up later:

https://github.com/ignitionrobotics/ign-gazebo/blob/8804138a7637150786e9618c851066d8cce6ab05/src/systems/joint_controller/JointController.cc#L191-L199

So I think that we should do one of the 2:

1. If there's no way the joint can show up later, we should remove the checks in `PreUpdate`, get `this->jointEntity` in `Configure` and return on failure.
1. If there's a way a joint can show up later, we shouldn't print an error at `Configure`.

I believe that `1.` covers the usual use case. Do you remember why those checks were added in the first place, @azeey ?

**Comment by cjds** [comment]
@chapulina I agree with `1.` but not necessarily `2.` Maybe we can print a warning instead? Whole reason I made the PR was I spent the day wondering why my joint wouldn't actuate only to find out it was a typo

**Comment by chapulina** [comment]
> Whole reason I made the PR was I spent the day wondering why my joint wouldn't actuate only to find out it was a typo

Ouch yeah trying infinitely silently is bad. Let's see what @azeey says, I think we could keep your error message and return right after it.

**Comment by azeey** [comment]
> There's some logic on PreUpdate to continuously look for the joint if it doesn't exist yet. This suggests to me that in some use cases, the joint may not be present on the model yet when Cofigure is called, but it may show up later:

hmm, I don't remember exactly why I added that check. Maybe I misunderstood how plugins are loaded. My current understanding is that for a given model, all the entities in the model are created before the plugin for that model is loaded. Waiting for entities is required only if that entity belongs to another model. So, option `1` would work, with the modification that we set a flag (eg. `validConfig = false`), so that when `PreUpdate` is subsequently called, we can return early by checking that flag.

**Comment by cjds** [comment]
Okay I think I understand that change. I'm going to go ahead and try to change my PR to be that instead

**Comment by ahcorde** [comment]
friendly ping @cjds

**Comment by ahcorde** [comment]
another friendly ping @cjds

**Comment by cjds** [comment]
Sorry @ahcorde  completely forgot about making changes. Just made the required changes.

**Comment by cjds** [comment]
@ahcorde I don't think I quite understand the code well enough to understand why the tests are failing.

I think it might be related to this line but maybe you could explain it to me
https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo5/test/integration/joint_controller_system.cc#L99

**Review by chapulina — APPROVED** [review]
I pushed some updates in e128aab5c7a2e124dd70c9204bd795e993424139:

* Removed `validConfig` variable because we can check if the system is valid by checking the `jointEntity`
* Demoted `jointName` to a local variable now that it's only used in one function
* Moved `kNullEntity` check to the top of `PreUpdate`
* Added a simple test to make sure nothing crashes
* 
LGTM with happy CI

**Review by ahcorde — APPROVED** [review]

**Pull request merged**